src/components/Autocomplete.tsx

```diff
@@ -0,0 +1,149 @@
+import React, { useCallback, useEffect, useRef, useState } from 'react';
+import { Person } from '../types/Person';
+
+interface IProps {
+  people: Person[];
+  onSelected?: (person: Person) => void;
+  onInputChange?: (inputValue: string) => void;
+}
+
+export const Autocomplete: React.FC<IProps> = React.memo(
+  ({ people, onSelected = () => {}, onInputChange = () => {} }) => {
+    const [inputText, setInputText] = useState('');
+    const [appliedInputText, setAppliedInputText] = useState('');
+    const [filteredPeople, setFilteredPeople] = useState<Person[]>([]);
+    const [showSuggestions, setShowSuggestions] = useState(false);
+
+    const timerId = useRef(0);
+
+    const useOutsideClick = (callBack: () => void) => {
+      const ref = React.useRef<HTMLInputElement>(null);
+
+      const handleClick = useCallback(
+        (event: MouseEvent) => {
+          if (ref.current && !ref.current.contains(event.target as Node)) {
+            callBack();
+          }
+        },
+        [callBack],
+      );
+
+      useEffect(() => {
+        document.addEventListener('click', handleClick);
+
+        return () => {
+          document.removeEventListener('click', handleClick);
+        };
+      }, [handleClick]);
+
+      return ref;
+    };
+
+    const handleClickOutside = () => {
+      setShowSuggestions(false);
+    };
+
+    const ref = useOutsideClick(handleClickOutside);
+
+    useEffect(() => {
+      if (appliedInputText === '') {
```

Comment:
```suggestion
      if (!appliedInputText) {
```

Reply:
![image](https://github.com/mate-academy/react_autocomplete/assets/55200530/26a633a9-9f85-42c0-a913-62afee5b2380)
still not updated